### PR TITLE
modules, pkgs: add rbrowser

### PR DESCRIPTION
**Pull request by rissson**
TODO:

* [x] test the module
* [ ] add profiles option
* [x] remove rbrowser's dependency to zsh
* [x] rbrowser shouldn't assume a default browser
* [x] https://github.com/kalbasit/nur-packages/pull/7

### CODE REVIEW

File: modules/programs/rbrowser/default.nix [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+          type = types.submodule browserSubmodule;
+        };
+        firefox = mkOption {
+          type = types.submodule browserSubmodule;
+        };
+      };
+
+      package = mkOption {
+        type = types.path;
+        default = pkgs.rbrowser.override {
```
Comment (kalbasit) [inline modules/programs/rbrowser/default.nix:66]: You should define `defaultText` as when the `default` points to a package.

File: modules/programs/rbrowser/default.nix [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        chromium = mkOption {
+          type = types.submodule browserSubmodule;
+        };
+        firefox = mkOption {
+          type = types.submodule browserSubmodule;
+        };
+      };
+
+      package = mkOption {
+        type = types.path;
```
Comment (kalbasit) [inline modules/programs/rbrowser/default.nix:65]:
```suggestion
        type = types.package;
```

File: pkgs/rbrowser/bin/rbrowser [hunk 3]
```diff
@@ -0,0 +1,162 @@
+#! @zsh_bin@
+#
+#  vim:ft=sh:
+#
+#  Copyright (c) 2010-2017 Wael Nasreddine <wael.nasreddine@gmail.com>
```
Comment (kalbasit) [inline pkgs/rbrowser/bin/rbrowser:5]:
```suggestion
#  Copyright (c) 2010-2020 Wael Nasreddine <wael.nasreddine@gmail.com>
```

File: pkgs/rbrowser/bin/rbrowser [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+#  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#  GNU General Public License for more details.
+#
+#  You should have received a copy of the GNU General Public License
+#  along with this program; if not, write to the Free Software
+#  Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307,
+#  USA.
+#
+
+set -eo pipefail
```
Comment (kalbasit) [inline pkgs/rbrowser/bin/rbrowser:23]:
```suggestion
set -euo pipefail
```

Please test with `-u` as it prevents access to undefined variables that may cause issues if not all of them are vetted.

File: pkgs/rbrowser/bin/rbrowser [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+			if [[ -d "${i}" ]]; then
+				items="${items}${sep}chromium@$(basename "${i}")"
+				sep="\n"
+			fi
+		done
+	fi
+	if [[ -d "$FIREFOX_PROFILES_PATH" ]]; then
+		for i in $FIREFOX_PROFILES_PATH/*; do;
+			if [[ -d "${i}" ]]; then
+				items="${items}${sep}firefox@$(basename "${i}")"
```
Comment (kalbasit) [inline pkgs/rbrowser/bin/rbrowser:105]:
```suggestion
				items="${items}${sep}firefox@$(basename "${i}")"
				sep="\n"
```

File: pkgs/rbrowser/default.nix [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+
+  src = ./.;
+
+  phases = [ "installPhase" "fixupPhase" ];
+
+  installPhase = ''
+    install -Dm755 $src/bin/rbrowser $out/bin/rbrowser
+
+    substituteInPlace $out/bin/rbrowser \
+      ${optionalString withBrave "--subst-var-by brave_bin ${brave}/bin/brave"} \
```
Comment (kalbasit) [inline pkgs/rbrowser/default.nix:36]: You're not really enabling brave here, you're only skipping setting the path to the brave binary. IMO bring the shell script inline so you can put the conditional around the entire brave-related sections of the code. Same applies to the rest.

**Review by kalbasit — CHANGES_REQUESTED** [review]

File: pkgs/rbrowser/bin/rbrowser (hunk 4 above)
Comment (rissson) [reply pkgs/rbrowser/bin/rbrowser:23]: This will require changing some logic in the arguments parsing though.

**Comment by rissson** [comment]
I think we can table the profiles option and removing the zsh dependency for later.

File: pkgs/rbrowser/default.nix [hunk 7]
```diff
@@ -0,0 +1,232 @@
...
+, zsh
+}:
+
+assert withBrave || withChromium || withFirefox;
+let
+  inherit (stdenvNoCC.lib) optionalString;
+
+
+  rbrowser = writeScriptBin "rbrowser" ''
+    #! ${zsh}/bin/zsh
```
Comment (kalbasit) [inline pkgs/rbrowser/default.nix:20]: Why zsh?

```suggestion
    #! ${runtimeShell}
```

File: pkgs/rbrowser/default.nix [hunk 8]
```diff
@@ -0,0 +1,232 @@
...
+    browser=""
+
+    while [[ $# -ge 1 ]]; do
+      case "''${1}" in
+        -h|--help)
+          help
+          exit 0
+          ;;
+        -p|--profile)
+          if [[ -z "''${2}" ]]; then
```
Comment (kalbasit) [inline pkgs/rbrowser/default.nix:65]:
```suggestion
          if [[ -z "''${2:-}" ]]; then
```
By setting an empty default, bash will do complaining about it being undefined. In talking about the set -u option

**Review by kalbasit — CHANGES_REQUESTED** [review]

**Review by kalbasit — APPROVED** [review]

**Pull request merged**